xtext/org.icyphy.linguafranca/src/org/icyphy/generator/CppGenerator.xtend

```diff
@@ -266,20 +266,20 @@ class CppGenerator extends GeneratorBase {
                 return port.widthSpec.terms.get(0).width
             }
         }
-        throw new Exception("Only integer valued multiport widths are supported for now.")
+        throw new Exception("Only multiport widths with literal integer values are supported for now.")
```

Comment:
Shouldn't this be checked by the validator rather than throwing an exception here?

Reply:
Good idea. Done. The TS target now fails to validate if you try to create multiports or banks. The Cpp target fails to validate if you try to set the width using a parameter.